### PR TITLE
fix(pubsublite)!: rename package ps to pscompat (api review)

### PR DESCRIPTION
**Pull request by tmdiep**
`pscompat` is more descriptive and indicates that this package is a compatibility API, designed to emulate the [Cloud Pub/Sub](https://pkg.go.dev/cloud.google.com/go/pubsub) API.

AI from API review.

### CODE REVIEW

**Review by hongalex — APPROVED** [review]
LGTM pending other approvals from API review attendees

**Review by codyoss — APPROVED** [review]
LGTM

**Review by tbpg — APPROVED** [review]

**Review by manuelmenzella-google — APPROVED** [review]

**Pull request merged**
